### DISCUSSION TITLE
Provide example on how to use next-safe-action / react hook form / shadcn

### DISCUSSION THREAD

**Discussion by bytechase** [next-safe-action/next-safe-action#265] [General]
Hey!

Just recently discovered this tool and already I am seeing the benefits and will be definitely implementing this moving forward for all server actions. I did notice in the docs you provide helpful documentation with react hook form, but do you have an example on how to use the helpful hooks with also the shadcn library?

Thanks!

**Comment by merthanmerter** [comment]
> Hey!
> 
> Just recently discovered this tool and already I am seeing the benefits and will be definitely implementing this moving forward for all server actions. I did notice in the docs you provide helpful documentation with react hook form, but do you have an example on how to use the helpful hooks with also the shadcn library?
> 
> Thanks!

Would this work for you?
[Next Safe Actions & React Hook Form - Reusable form component](https://gist.github.com/merthanmerter/beaafd1602873aac5aef457f0796a808)
